### DISCUSSION TITLE
Snoop for Avalonia

### DISCUSSION THREAD

**Discussion by npolyak** [AvaloniaUI/Avalonia#6126] [Q&A] [answered]
Is there a tool like Snoop for Avalonia - a tool that would allow to check the visual tree, to check all non-default Attached properties, to change attached properties, to check the DataContext, to call methods on an element navigated to.

Thanks

**Comment by maxkatz6** [comment, marked as answer]
Avalonia has DevTools build in. Press F12 while app is running.
In default templates of your window's .xaml.cs file you can find AttachDevTools method invoked, which sets everything app to enable dev tools for that specific window.

**Reply by maxkatz6** [reply]
We have small documentation about it as well https://docs.avaloniaui.net/docs/getting-started/developer-tools

**Reply by npolyak** [reply]
Cool, Thanks!

**Reply by geocine** [reply]
Do you have plans to be able to drill down to the `DataContext` and edit accordingly?

**Reply by maxkatz6** [reply]
> Do you have plans to be able to drill down to the DataContext

Yes. I think in nightly builds.

> and edit accordingly

Not sure, most likely limited

**Comment by liwuqingxin** [comment]
A commercial devtools for avalonia is available now: [DevTools for Avalonia](https://www.devtools.nlnet.net)
